fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeActivity.java

```diff
@@ -244,12 +246,34 @@ public void setupHomeButton() {
 
   public void setupConnectButton() {
     Button connectButton = findViewById(R.id.tab_connect);
-    connectButton.setOnClickListener(v -> mViewModel.openConnect());
+    connectButton.setOnClickListener(v -> {
+        if (mViewModel != null && !mViewModel.isWalletSetUp()){
```

Comment:
Can the view model be null ?
That's surprising

Reply:
It was a code smell according to SonarCloud, if I remember correctly. Worst case scenario it does no harm. Should I remove it ?

Reply:
This is probably a false positive as the view model is initialized in the onCreate() function.

Reply:
Maybe I remove it then, it could cause confusion to a future inexperienced student looking at that. What do you think ?

Reply:
That is a good point.
You can also report this as a false positive in sonar.